src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -1,4 +1,44 @@
 package com.dev.cinema.dao.impl;
 
-public class UserDaoImpl {
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.exceptions.DataProcessingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+import java.util.Optional;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add new user", e);
```

Comment:
Mentors recommended making this message more informative
```suggestion
            throw new DataProcessingException("Can't add new " + user, e);
```